### ISSUE TITLE
SIM115 false positive on functions that return file handler objects

### ISSUE THREAD

**Issue by gboeing** [astral-sh/ruff#13862]
ruff 0.7 gives a false positive for SIM115 on something like:

```python
def my_function(filepath, encoding):
    return bz2.open(filepath, mode="rt", encoding=encoding)
```

which can be resolved like:
```python
def my_function(filepath, encoding):
    with bz2.open(filepath, mode="rt", encoding=encoding) as f:
        return f
```

But this prevents the calling function from being able to work with `f` because the context manager has closed it. It seems that SIM115 should not trigger when returning a file handler object from a function?

**Comment by zanieb** [comment]
Yeah strong agree that this should not be flagged.

Related https://github.com/astral-sh/ruff/pull/13679#issuecomment-2400078411

**Comment by AlexWaygood** [comment]
I agree that this could be improved. FWIW, though, this is really a pre-existing issue that could occur prior to Ruff 0.7. For example, Ruff 0.1 emits `SIM115` on this code that uses the builtin `open` function:

```py
def my_function(filepath, encoding):
    return open(filepath, mode="rt", encoding=encoding)
```

The only difference in Ruff 0.7 is that the rule is now applied consistently to more stdlib functions that open files, rather than just the builtin `open` function and `pathlib.Path.open`.

**Comment by AlexWaygood** [comment]
> ```python
> def my_function(filepath, encoding):
>     return bz2.open(filepath, mode="rt", encoding=encoding)
> ```

Ideally, of course, rather than writing a wrapper function that _returns_ an open file handle, you'd write a wrapper context manager that _yields_ an open file handle, because then you can be confident that the file will always be closed after you're done with it:

```python
from contextlib import contextmanager

@contextmanager
def my_function(filepath, encoding):
    with bz2.open(filepath, mode="rt", encoding=encoding) as f:
        yield f
```

Enforcing that kind of practice is probably outside the scope of this rule, however.
